Feature 1852 use case physics tendency

bikegeek opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

1. on host 'seneca', using met_test conda environment metplotpy.v5:

 _conda activate /home/met_test/.conda/envs/metplotpy.v5_

2. Clone the METplus feature_1852_useCase_physics_tendency repo

3. Under the same 'base' directory as the METplus repo, clone the METplotpy 'develop' branch (METplus and METplotpy will be at the same directory level)
4. Obtain the short_range data tarball from host 'mohawk':
    /d2/www/dtcenter/dfiles/code/METplus/METplus_Data/v5.0/sample_data-short_range-5.0.tgz
5. Uncompress the tarball 
6. Set the following environment variables on 'seneca' :

- MET_BUILD_BASE (set to /usr/local/met-10.0.0 for 'seneca')
- METPLUS_DATA (set to location of uncompressed data in step 5)
- METPLUS_BASE (set to location of the METplus source code)

7. Create user_system.conf file and set the:

- OUTPUT_BASE
- INPUT_BASE = {ENV[METPLUS_DATA]}
- MET_INSTALL_DIR = /usr/local/met-10.0.0 

8. Generate the **planview plot**:
_**python run_metplus.py 
path/to/parm/use_cases/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency_Planview.conf /your/path/to/user_system.conf**_

In the OUTPUT_BASE/plots directory, you will find the _short_range-physics_tendency_planview.png_

![Screen Shot 2022-10-20 at 10 25 22 AM](https://user-images.githubusercontent.com/3753118/197005254-33c16ea1-caa0-48b7-a893-ecc9679dfde0.png)


9. Repeat the above steps to generate the **Vertical Profile plot**, except use this conf file with the same user_system.conf file:
**UserScript_fcstFV3_fcstOnly_PhysicsTendency_VerticalProfile.conf**

In the OUTPUT_BASE/plots directory, you will find the _short_range-physics_tendency_vertical_profile.png_ 

![Screen Shot 2022-10-20 at 10 28 02 AM](https://user-images.githubusercontent.com/3753118/197005904-07bb7c7c-c5f7-4941-8f68-4801dbc64209.png

10. Generate the **Vertical Cross Section plots** by repeating, except use this conf file:
**UserScript_fcstFV3_fcstOnly_PhysicsTendency_VerticalCrossSection.conf**
with the same user_system.conf file.  

You will find the _short_range-physics_tendency_vertical_cross_section.png_ in the OUTPUT_BASE/plots directory.

![Screen Shot 2022-10-20 at 10 32 16 AM](https://user-images.githubusercontent.com/3753118/197006532-203d1579-3a74-48e4-9dbb-9afdae207f61.png)



- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Perform same steps as above

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes ]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[Yes]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br> New data

- [x] Please complete this pull request review by **[Earliest convenience to meet beta4 release]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Development** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

bikegeek committed:
Issue #1852 initial work for the Planview plot

bikegeek committed:
Issue #1852 planview plot for physics tendency

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METplus into feature_1852_useCase_physics_tendency

bikegeek committed:
issue #1852 cleared away unused/unneccessary code

bikegeek committed:
Issue #1852 update script command

bikegeek committed:
Issue #1852 first draft, doesn't have correct content

bikegeek committed:
Merge branch 'feature_1852_useCase_physics_tendency' of https://github.com/dtcenter/METplus into feature_1852_useCase_physics_tendency

bikegeek committed:
Issue #1852 updated name of output plot to include short_range in name

bikegeek committed:
Issue #1852 renamed to include fcstOnly label

bikegeek committed:
Issue #1852 initial commit

bikegeek committed:
Issue #1852

bikegeek committed:
Issue #1852

bikegeek committed:
Issue #1852 initial commit

bikegeek committed:
issue #1852 vertical cross section

bikegeek committed:
Issue 1852 clean up unused lines

bikegeek committed:
Issue #1852 initial commit

bikegeek committed:
Issue #1852

bikegeek committed:
Issue #1852 updates

bikegeek committed:
Issue #1852 fixed documentation warnings, but some problems remain

lisagoodrich committed:
fixing line length for chapter and section headers and one letter capitalization

lisagoodrich committed:
cleaning up line spacing and trying to fix a critical error.

lisagoodrich committed:
trying to fix critical error again

bikegeek committed:
Issue #1852 more description for data and dependency on METplotpy with additional instructions

bikegeek committed:
Issue #1852 update location of input data to single directory

Commit:
Removing duplicate files with a misformatted name that are not needed

Commit:
Merge branch 'feature_1852_useCase_physics_tendency' of github.com:dtcenter/METplus into feature_1852_useCase_physics_tendency

Commit:
Per #1852, making changes to formatting to eliminate errors

bikegeek committed:
Issue #1852 remove redundant yaml config file

bikegeek committed:
Merge branch 'feature_1852_useCase_physics_tendency' of https://github.com/dtcenter/METplus into feature_1852_useCase_physics_tendency

bikegeek committed:
Issue #1852 update PYTHONPATH information

bikegeek committed:
Merge branch 'feature_1852_useCase_physics_tendency' of https://github.com/dtcenter/METplus into feature_1852_useCase_physics_tendency

bikegeek committed:
Issue #1852 fixed conflicts

bikegeek committed:
Issue #1852 update path to shapefile

bikegeek committed:
Issue #1852 fixed copy and paste error

bikegeek committed:
Merge branch 'feature_1852_useCase_physics_tendency' of https://github.com/dtcenter/METplus into feature_1852_useCase_physics_tendency

bikegeek committed:
Issue #1852 add three use case entries: Planview, VerticalProfile, and VerticalCrossSection for the short range Physics tendency use case

bikegeek committed:
Issue #1852 add latest short range tests to top, move down list when everything passes

bikegeek committed:
Update use_case_groups.json

move short_range use cases #10-12 to below the short range test for #9

georgemccabe reviewed (approved):
I am just reviewing the use case config format and execution in the automated tests. @fisherhucar will review the use case documentation.

I did not run the use cases on seneca, but I reviewed the use case runs in GitHub Actions and confirmed that they ran successfully. I downloaded the output data generated from the use cases and confirmed that the images look good. I confirmed that the only differences (besides other use case groups' output that were recently updated since this PR) are due to new output images created by the use cases, which is expected.

The use case config files are well organized based on the new ordering structure. My only suggestion would be to pull out arguments found in the USER_SCRIPT_COMMAND and values set in the YAML file that a user may want to adjust into variables in the METplus config file and reference those variables in the command and YAML. This is optional, but would be nice to add at some point so that it is easier for users to take this example and adjust it as needed.

For example, if fv3_history.nc file is an input to the script that a user may want to change, you could put that into a variable and reference it in the user script command. You could also pull out the pressure level (assuming -p 500 sets the pressure level to use?).

Before:

`USER_SCRIPT_COMMAND = {PARM_BASE}/use_cases/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency_Planview/planview_plot.py {PARM_BASE}/use_cases/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency_Planview/physics_tendency_planview.yaml {INPUT_BASE}/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency/fv3_history.nc {INPUT_BASE}/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency/grid_spec.nc tmp pbl -p 500 -t 1 -v 20190504T14 -o {OUTPUT_BASE}/plots/short_range-physics_tendency_planview.png --nofineprint`

After:
```
FV3_HISTORY_FILE = {INPUT_BASE}/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency/fv3_history.nc
PRESSURE_LEVEL = 500

USER_SCRIPT_COMMAND = {PARM_BASE}/use_cases/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency_Planview/planview_plot.py {PARM_BASE}/use_cases/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency_Planview/physics_tendency_planview.yaml {FV3_HISTORY_FILE} {INPUT_BASE}/model_applications/short_range/UserScript_fcstFV3_fcstOnly_PhysicsTendency/grid_spec.nc tmp pbl -p {PRESSURE_LEVEL} -t 1 -v 20190504T14 -o {OUTPUT_BASE}/plots/short_range-physics_tendency_planview.png --nofineprint
```

Again this is optional cleanup and isn't required for this PR. I approve.

georgemccabe commented:
Note that after this PR is merged, we will need to update the develop-ref branch to include the new output that is generated by these 3 use cases.

bikegeek committed:
Issue #1852 clean up conf files by providing variables for the input netCDF files and for the planview plot, allow user to indicate pressure levels

bikegeek committed:
Issue #1852 Indicate external Python packages

bikegeek committed:
Issue #1852 added external dependency section

georgemccabe committed:
Merge branch 'develop' into feature_1852_useCase_physics_tendency

georgemccabe committed:
removed env vars that don't need to be set -- METPLOTPY_BASE is set by the automated test conf (.github/parm/test_settings.conf) and PYTHONPATH doesn't need to be set because METplotpy is installed in Python environment for use case tests

hankenstein2 committed:
Update UserScript_fcstFV3_fcstOnly_PhysicsTendency_Planview.py

Fixed one typo

hankenstein2 committed:
Update UserScript_fcstFV3_fcstOnly_PhysicsTendency_VerticalCrossSection.py

Fixed a typo

hankenstein2 committed:
Update UserScript_fcstFV3_fcstOnly_PhysicsTendency_VerticalProfile.py

Same typo, same fix

hankenstein2 reviewed (approved):
Everything looks good documentation wise.

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METplus into feature_1852_useCase_physics_tendency

bikegeek committed:
Merge branch 'feature_1852_useCase_physics_tendency' of https://github.com/dtcenter/METplus into feature_1852_useCase_physics_tendency

bikegeek committed:
Issue #1852 remove extraneous entry for USER_SCRIPT_COMMAND, a hold-over from the original implementation which doesn't allow the user to easily specify the FV3 history file and grid spec netCDF file

bikegeek committed:
Update UserScript_fcstFV3_fcstOnly_PhysicsTendency_VerticalProfile.conf

Removed the incorrect USER_SCRIPT_COMMAND in previous commit.  Reinstated the correct version and removed the incorrect command.

bikegeek committed:
update html file from Planview to VerticalProfile
